backend/app/main.py

```
@@ -215,7 +216,33 @@ async def cleanup_old_diagnosis_images():
+# progress notification 은 단일 tool 호출 내부 SSE 스트림으로 그대로 동작하므로
+# T4 (analyze_reviews_with_progress) 도 영향 없음. cross-request 상태 공유만 비활성화.
+_review_mcp_app = _review_mcp.http_app(path="/", stateless_http=True)
+
+try:
+    from fastmcp.utilities.lifespan import combine_lifespans
+    _combined_lifespan = combine_lifespans(lifespan, _review_mcp_app.lifespan)
+except ImportError:
+    # combine_lifespans 가 없는 fastmcp 버전일 때의 폴백.
+    # 두 lifespan 을 직접 nested asynccontextmanager 로 합성한다.
+    @asynccontextmanager
+    async def _combined_lifespan(app: FastAPI):  # type: ignore[no-redef]
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_combined_lifespan`

(ANN202)